lumen/ai/agents.py

```diff
@@ -903,6 +916,44 @@ async def _update_spec(self, memory: _Memory, event: param.parameterized.Event):
             return
         memory['view'] = dict(spec, type=self.view_type)
 
+    def _add_geographic_items(self, vega_spec: dict, vega_spec_str: str):
+        # standardize the vega spec, by migrating to layer
+        if "layer" not in vega_spec:
+            vega_spec["layer"] = [{"encoding": vega_spec.pop("encoding", None), "mark": vega_spec.pop("mark", None)}]
+
+        # make it zoomable
+        if "params" not in vega_spec:
+            vega_spec["params"] = []
+        vega_spec["params"].extend(VEGA_ZOOMABLE_MAP_ITEMS["params"])
+        if "projection" not in vega_spec:
+            vega_spec["projection"] = {"type": "mercator"}
+        vega_spec["projection"].update(VEGA_ZOOMABLE_MAP_ITEMS["projection"])
+
+        # add outlines
+        if "world-110m.json" in vega_spec_str and vega_spec["projection"]["type"] == "albersUsa":
+            vega_spec["projection"] = "mercator"  # cannot use albersUsa with world-110m
+        elif "world-110m.json" not in vega_spec_str and vega_spec["projection"]["type"] != "albersUsa":
```

Comment:
Confused by these if/else clauses (actually most the `elif` part). Also it seems fairly presumptive to add a US layer automatically.

Reply:
first if: using albersUsa with coastlines causes warping
<img width="674" alt="image" src="https://github.com/user-attachments/assets/694c2d09-c05c-4cc5-8820-2dd2e260cb9b" />

second elif, if no coastline and the projection isn't albersUsa, add a coastline layer

third elif, if no us states but projection is albersUsa, add a states layer (assuming no other coastline layer, which would be caught in first if)